### PR TITLE
fix(ecs): DeploymentAlarms property is specified for ECS service with CODE_DEPLOY and EXTERNAL deployment controller

### LINKED ISSUE #26307: ECS Fargate Service: service not created because of alarm based rollback
### Describe the bug

During the deployment process with the same code which was working previously is returning the following error.

```
4:43:04 pm | CREATE_FAILED        | AWS::ECS::Service                               | xxxx
Resource handler returned message: "Invalid request provided: CreateService error: Alarm based rollback feature is only supported with ECS deployment controller. Update t
o ECS deployment controller and try again. (Service: AmazonECS; Status Code: 400; Error Code: InvalidParameterException; Request ID: xxx;
Proxy: null)" (RequestToken: xxx, HandlerErrorCode: InvalidRequest)
```

The following is my ECS deployment group
```
const ecsDeploymentGroup = new codeDeploy.EcsDeploymentGroup(this, 'ecsDeploymentGroup', {
      application: ecsApplication,
      deploymentGroupName: props.ECSTaskFamilyName,
      deploymentConfig,
      alarms: ecsServiceAlarms.alarms,
      service: ecsBlueGreenService.ecsService,
      blueGreenDeploymentConfig: {
        blueTargetGroup: ecsBlueGreenService.blueTargetGroup,
        greenTargetGroup: ecsBlueGreenService.greenTargetGroup,
        listener: ecsBlueGreenService.albProdListener,
        testListener: ecsBlueGreenService.albHttpsListener,
        terminationWaitTime: Duration.minutes(10),
      },
      autoRollback: {
        stoppedDeployment: true,
      }
    });
```

The application is deployed using Code deploy, So the deployment controller is set as CODE_DEPLOY

```
this.ecsService = new ecs.FargateService(this, 'FargateService', {
      cluster: props.cluster!,
      desiredCount: 1,
      taskDefinition,
      minHealthyPercent: 100,
      maxHealthyPercent: 200,
      enableExecuteCommand: true,
      deploymentController: {
        type: DeploymentControllerType.CODE_DEPLOY
      },
      healthCheckGracePeriod: Duration.seconds(60)
    });

```

### Expected Behavior

ECS Cluster gets deployed successfully

### Current Behavior

ECS creation failed

### Reproduction Steps

Use the code in the bug description 

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.87.0

### Framework Version

_No response_

### Node.js Version

v18.15.0

### OS

MacOS 14.0

### Language

Typescript

### Language Version

5.1.6

### Other information

_No response_

### PR DESCRIPTION
**Pull request by tam0ri**
From #25840, ECS L2 construct sets the default configuration for the `CfnService.deploymentConfiguration.alarms` property to:
```
alarmNames: [],
rollback: false,
enable: false,
```

However, alarm based rollback feature is only supported for ECS services that use the rolling update (ECS) deployment controller.
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/deployment-alarm-failure.html

Due to this limitation, when deploymentController is set to CODE_DEPLOY or EXTERNAL, creation for the service will fail by conflict with `deploymentConfiguration.alarms` property.

This PR solves the issue by skipping to set default configuration for the `CfnService.deploymentConfiguration.alarms` property for CODE_DEPLOY and EXTERNAL deployment controller.

Closes #26307

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by tam0ri** [comment]
Exemption Request for this quick fix. If integration tests and snapshot are still needed, please let me know.

**Comment by tam0ri** [comment]
I checked failed build. I want to take a new snapshot for @aws-cdk-testing
/framework-integ/test/aws-codedeploy/test/ecs/integ.deployment-group.ts, but I can't execute integ-runner now. `npx lerna run build --scope=@aws-cdk-testing/framework-integ` failed with the following error.

```
$ cdk-build
       
       /workspace/aws-cdk/packages/aws-cdk/lib/commands/migrate.ts
         5:24  error  Unable to resolve path to module '../vendor/noctilucent'  import/no-unresolved
       
       ✖ 1 problem (1 error, 0 warnings)
       
       Error: /workspace/aws-cdk/node_modules/eslint/bin/eslint.js . --ext=.ts --resolve-plugins-relative-to=/workspace/aws-cdk/tools/@aws-cdk/cdk-build-tools/lib exited with error code 1
```

I usually use devcontainer, so I tried to use Gitpod, but the problem does not solved.

**Comment by mrgrain** [comment]
@tam0ri Is your branch up-to-date? We made a change to how we import `noctilucent` recently. 
Also did you receive any errors during the build?

File: packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts [hunk 1]
```diff
@@ -682,7 +682,9 @@ export abstract class BaseService extends Resource
         enable: true,
         rollback: props.deploymentAlarms.behavior !== AlarmBehavior.FAIL_ON_ALARM,
       };
-    } else if (this.deploymentAlarmsAvailableInRegion()) {
+    // CloudWatch alarms is only supported for Amazon ECS services that use the rolling update (ECS) deployment controller.
+    } else if (props.deploymentController?.type !== DeploymentControllerType.CODE_DEPLOY &&
+      props.deploymentController?.type !== DeploymentControllerType.EXTERNAL && this.deploymentAlarmsAvailableInRegion()) {
```
Comment (mrgrain) [inline packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts:687]: Any particular reason you are checking for the negative here (not CODE_DEPLOY, not EXTERNAL) and not if the controller is ECS?

**Review by mrgrain — CHANGES_REQUESTED** [review]
Code looks good. Thank you! Just the one small question. If you add the test I can run the snapshots for you.

**Comment by tam0ri** [comment]
I pushed the fix. If you have any comment, please feel free to point it out!

> Is your branch up-to-date? We made a change to how we import noctilucent recently.

Yes, I already pulled the latest main branch. I'll dig it further more later, but it really helpful that you add a new snapshot.

**Comment by tam0ri** [comment]
I could execute integ-runner on Gitpod, so I added a new snapshot by myself.

The previous snapshot includes the invalid configuration (alarm based rollback + CODE_DEPLOY deployment controller).
https://github.com/aws/aws-cdk/blob/9fca7909797e71788b65bd1bf7bd2c0c85a75159/packages/%40aws-cdk-testing/framework-integ/test/aws-codedeploy/test/ecs/integ.deployment-group.js.snapshot/aws-cdk-codedeploy-ecs-dg.template.json#L445-L462

So, I created the new snapshot with `--disable-update-workflow`.

**Review by mrgrain — APPROVED** [review]

**Pull request merged**